test/with_dummyserver/test_socketlevel.py

```diff
@@ -911,6 +911,54 @@ def echo_socket_handler(listener):
         r = conn.urlopen('GET', url, retries=0)
         self.assertEqual(r.status, 200)
 
+    def test_connect_ipv6_addr(self):
+        ipv6_addr = '2001:4998:c:a06::2:4008'
+
+        def echo_socket_handler(listener):
+            sock = listener.accept()[0]
+
+            buf = b''
+            while not buf.endswith(b'\r\n\r\n'):
+                buf += sock.recv(65536)
+            s = buf.decode('utf-8')
+            if s.startswith('CONNECT %s:443' % (ipv6_addr,)):
+                sock.send(b'HTTP/1.1 400 Bad Request\r\n\r\n')
+                sock.close()
+                return
```

Comment:
Should I just `else` statement to close the socket? I am not sure what `elif` condition you would want me check. For example:

```python
        def echo_socket_handler(listener):
            sock = listener.accept()[0]

            buf = b''
            while not buf.endswith(b'\r\n\r\n'):
                buf += sock.recv(65536)
            s = buf.decode('utf-8')
            if s.startswith('CONNECT [%s]:443' % (ipv6_addr,)):
                sock.send(b'HTTP/1.1 200 Connection Established\r\n\r\n')
                ssl_sock = ssl.wrap_socket(sock,
                                           server_side=True,
                                           keyfile=DEFAULT_CERTS['keyfile'],
                                           certfile=DEFAULT_CERTS['certfile'])
                buf = b''
                while not buf.endswith(b'\r\n\r\n'):
                    buf += ssl_sock.recv(65536)

                ssl_sock.send(b'HTTP/1.1 200 OK\r\n'
                              b'Content-Type: text/plain\r\n'
                              b'Content-Length: 2\r\n'
                              b'Connection: close\r\n'
                              b'\r\n'
                              b'Hi')
                ssl_sock.close()
            else:
                socket.close()
```

Do you have something else in mind?